staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -166,3 +166,56 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FlowControlConfig holds flow control parameters.
+type FlowControlConfig struct {
+	metav1.TypeMeta
+
+	// WorkEstimatorConfig holds work estimator parameters.
+	// +optional
+	*WorkEstimatorConfig
+}
+
+// WorkEstimatorConfig holds work estimator parameters.
+type WorkEstimatorConfig struct {
+	// ListWorkEstimatorConfig holds work estimator parameters related to list requests.
+	// +optional
+	ListWorkEstimatorConfig *ListWorkEstimatorConfig
+	// MutatingWorkEstimatorConfig holds work estimator
+	// parameters related to watches of mutating objects.
+	// +optional
+	MutatingWorkEstimatorConfig *MutatingWorkEstimatorConfig
+
+	// MaximumSeats is the maximum number of seats a request can occupy
+	//
+	// NOTE: work_estimate_seats_samples metric uses the value of maximumSeats
+	// as the upper bound, so when we change maximumSeats we should also
+	// update the buckets of the metric.
+	// +optional
+	MaximumSeats uint
+}
+
+// ListWorkEstimatorConfig holds work estimator parameters related to list requests.
+type ListWorkEstimatorConfig struct {
+	// ObjectsPerSeat is a number of objects from a single list request per seat.
+	// +optional
+	ObjectsPerSeat float64
+}
+
+// MutatingWorkEstimatorConfig holds work estimator
+// parameters related to watches of mutating objects.
+type MutatingWorkEstimatorConfig struct {
+	// Enabled enables mutating work estimator.
+	// TODO(wojtekt): Remove it once we tune the algorithm to not fail
+	// scalability tests.
+	// +optional
+	Enabled bool
+	// Additional duration a seat allocated to a watch event is reserved for.
+	// +optional
+	EventAdditionalDuration metav1.Duration
+	// WatchesPerSeat is a number of watches of a single event per seat.
```

Comment:
Since we are promoting this to an API, I think these comments deserve more care than they originally got.  For this one, perhaps something like the following would be better.

```
// `watchesPerSeat` is the ratio between the number of WATCHes that will get notified and
// the number of seats occupied for `eventAdditionalDuration` to account for the work of
// sending those notifications.
```

Reply:
Thanks, changed.